Makefile

```diff
@@ -23,14 +30,40 @@ e2e: deploy
 $(TARGETS): vendor/modules.txt
 	./scripts/$@
 
-build-subctl: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go")
+clean:
+	rm -f $(BINARIES)
 
-bin/subctl: build-subctl
+build: operator-image binaries
 
-pkg/subctl/operator/common/embeddedyamls/yamls.go: pkg/subctl/operator/common/embeddedyamls/generators/yamls2go.go $(shell find deploy/ -name "*.yaml")
-	$(MAKE) generate-embeddedyamls
+build-cross: $(CROSS_BINARIES)
 
-.PHONY: $(TARGETS)
+binaries: $(BINARIES)
+
+operator-image: vendor/modules.txt
+	operator-sdk build quay.io/submariner/submariner-operator:$(VERSION) --verbose
+
+bin/subctl: bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
+	ln -sf $(<F) $@
+
+bin/subctl-%: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go") vendor/modules.txt
+	version=$(word 2,$(subst -, ,$@)); \
+	GOOS=$(word 3,$(subst -, ,$@)); \
+	GOARCH=$(word 1,$(subst ., ,$(word 4,$(subst -, ,$@)))); \
+	GOEXE=$(word 2,$(subst ., ,$(word 4,$(subst -, ,$@)))); \
+	[ "$$GOOS" != "darwin" ] || upxflag="--noupx"; \
+	export GOARCH GOEXE GOOS; \
+	$(SCRIPTS_DIR)/compile.sh \
+		--ldflags "-X github.com/submariner-io/submariner-operator/pkg/version.Version=$${version} -X github.com/submariner-io/submariner-operator/pkg/version.GitRef=$(GIT_REF)" \
```

Comment:
The idea is to explicitly use the version extracted from the target name, without forcing it to be equal to `$(VERSION)`. Think of this as a function which is only supposed to act on its arguments, not on a global variable ;-).

Reply:
Ok sure, BTW we need to make sure VERSION has no dashes or this'll break

Reply:
We already rely on that, the version is embedded in the current binary and tarball names.

Reply:
Yeah but so far we didnt try to parse the tile name, now with 
```
 	version=$(word 2,$(subst -, ,$@)); \ 

	GOOS=$(word 3,$(subst -, ,$@)); \
	GOARCH=$(word 1,$(subst ., ,$(word 4,$(subst -, ,$@)))); \
	GOEXE=$(word 2,$(subst ., ,$(word 4,$(subst -, ,$@)))); \ 
```
You're really relying on the filename, and hence if for example version was `foo-bar` these will break as the calculations will take `foo` as the 2nd word and `bar` as the 3rd (and skew everything else, of course).

Actually now that I think of it, it'll  break right now since our calculation is a tag or `dev` buuut if you tag say `v0.4.0-rc1` (which we did and do) then these will break.

Or am I missing something?

Reply:
Oh I read slashes, not dashes, duh. You’re right, I need to find a better way to do this...

Reply:
I’ve updated the patch to rely on `$(VERSION)` and count backwards when splitting.